setup.py

```diff
@@ -117,13 +118,66 @@ def get_setup_kwargs(raw=False):
     return kwargs
 
 
+DEBUG_STRING = """
+            =============================DEBUG ASSISTANCE==========================
+            {}
+            =============================DEBUG ASSISTANCE==========================
+            """
+
+
+def check_namespace_clashes(raw=False):
+    # This runs before the setup, trying to catch common setup errors
+    try:
+        get_distribution("telegram")
+        print(
+            DEBUG_STRING.format(
+                """You are trying to install the `python-telegram-bot` library,
+            but the `telegram` library is already installed.
+            Both libraries provide a Python package called `telegram`, 
+            which will lead to a namespace clash.
+            If you want to use the `python-telegram-bot` library, 
+            please first uninstall the `telegram` library 
+            before installing the `python-telegram-bot` library again."""
+            )
+        )
+        raise Exception("BuildError")
+    except DistributionNotFound:
+        pass
+    try:
+        get_distribution(f'python-telegram-bot{"" if raw else "-raw"}')
+        print(
+            DEBUG_STRING.format(
+                f"""You can not use python-telegram-bot and python-telegram-bot-raw.
+            Please uninstall python-telegram-bot{"" if raw else "-raw"} and try again."""
+            )
+        )
+        raise Exception("BuildError")
+    except DistributionNotFound:
+        pass
+    try:
+        # if the telegram import works here, something locally overrides telegram
+        import telegram
```

Comment:
just realized, do we need this check? A user could `pip install` from any directory.